### PR TITLE
Allow removal of location and description via edit menu (#404)

### PR DESCRIPTION
**Pull request by hammerheadlemon**
This PR addresses #404 whereby users are unable to remove a description or location associated with an event via `gcalcli edit`. Entering a "" or " " (by pressing the return key at the prompt) moves the prompt onto the next question and does touch the `event` object.

This PR uses the convention whereby a user may enter a "." to clear the current contents of the field - I am presuming that a "." is an unlikely realistic candidate for data in either of these fields.

I have added tests to test for the both the removal and editing of data in the description and location field. The current state of `gcalcli.py` does not lend itself to be easily tested, so the tests aren't as concise as I'd like them to be, but I've tried to document what's going on in each and explain where things are being patched and why.  I've used a temporary environment variable (`GCALCLI_TEST_CYCLE`) in both the test and production code to order to properly control the tests and ensure they do not go into an infinite loop.  There will be a more elegant way, but without breaking up the long `elif val.lower() = "?"` structure in `gcalcli.py`, this is as good a way as any - and it's better to have tests than not.

Happy to take comments and feedback.

### CODE REVIEW

File: gcalcli/gcalcli.py [hunk 1]
```diff
@@ -105,6 +106,30 @@
...
+    We need to be able to patch input() funcs at various stages of testing.
+    This input is used to capture description choice when doing gaclcli edit
+    """
+    return get_input(printer, msg, func)
+
+
+def edit_location_input(printer, msg, func):
+    """
+    We need to be able to patch input() funcs at various stages of testing.
+    This input is used to capture description choice when doing gaclcli edit
```
Comment (hammerheadlemon) [inline gcalcli/gcalcli.py]: I need to fix the comment to refer to location.

**Comment by jcrowgey** [comment]
Thanks for this PR.  Take a look at my comment in #404 and let's think about whether we want to realign the usage of '.' in the add command.

**Comment by hammerheadlemon** [comment]
I'll make the changes mentioned in #404 and update.

**Comment by hammerheadlemon** [comment]
Updating PR - have moved tests for `gaclcli edit` into own module. Not looked at #404 yet.

**Pull request closed without merging**